org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/DOMCompletionContext.java

```diff
@@ -0,0 +1,68 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ *     Gayan Perera - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.internal.codeassist;
+
+import java.util.Collection;
+
+import org.eclipse.jdt.core.CompletionContext;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.dom.IBinding;
+
+class DOMCompletionContext extends CompletionContext {
+    private int offset;
+    private char[] token;
+    private IJavaElement enclosingElement;
+    private Collection<? extends IBinding> visibleBindings;
+
```

Comment:
Can you please make fields `final` as much as possible? I found it really prevents bugs from popping up in code.